### PR TITLE
build: add ability to download cached dist folder

### PR DESCRIPTION
**Pull request by HashimotoYT**
This PR adds support to download the cached dist folder for the future using in tests without needing to building big JSON files.

## Using

You need to provide a GitHub token, after that you can use `yarn get-cache` to download cache.

If you member of @electron org, the token what's are you provided should have enabled SSO. Otherwise, you get an error 🙈.

### CODE REVIEW

File: script/get-cache.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import { Octokit } from '@octokit/rest'
+import * as fs from 'fs'
+import * as path from 'path'
+import { tmpdir } from 'os'
+import * as extractZip from 'extract-zip'
+import got from 'got'
+import { ProxyStream } from 'got/dist/source/as-stream'
```
Comment (malept) [inline script/get-cache.ts]:
```suggestion
import * as extractZip from 'extract-zip'
import * as fs from 'fs'
import got from 'got'
import { Octokit } from '@octokit/rest'
import * as path from 'path'
import { promisify } from 'util'
import { ProxyStream } from 'got/dist/source/as-stream'
import * as stream from 'stream'
import { tmpdir } from 'os'

const pipeline = promisify(stream.pipeline)
```

File: script/get-cache.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        writeStream.destroy(error)
+      }
+
+      reject(error)
+    })
+    writeStream.on('error', (error) => reject(error))
+    writeStream.on('close', () => resolve())
+
+    downloadStream.pipe(writeStream)
+  })
```
Comment (malept) [inline script/get-cache.ts]:
```suggestion
  const downloadStream: ProxyStream<any> = got.stream(url, {
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${token}`,
    },
  })

  downloadStream.on('response', () => {
    // Artifacts doesn't have in the response `content-length` so we unable
    // to give back the progress bar to give more perfect response.
    // It's just simple way to say we just started downloading.
    console.log(`Downloading started.`)
  })
  await pipeline(downloadStream, writeStream)
```

**Comment by HashimotoYT** [comment]
> IMO, promisifying stream.pipeline is a better way to deal with piping streams with Node 10+, see the suggestions below.

I took that code from `@electron/get`, since not very many times worked with streams. Thanks for suggestions 🌻.

> (Side note: I'm really not a fan of @octokit/rest 17 with TypeScript at the moment...)

I dropped back to Octokit 16 and use the `octokit.request` function to create requested API.

File: package.json [hunk 3]
```diff
@@ -39,15 +39,16 @@
...
     "npm-run-all": "^4.1.2",
     "prettier": "^2.0.0",
+    "remark": "^11.0.2",
+    "remark-inline-links": "^4.0.0",
     "require-yaml": "^0.0.1",
     "rimraf": "^3.0.0",
     "roggy": "^0.1.0",
     "semantic-release": "^17.0.3",
     "simple-statistics": "^7.0.2",
+    "extract-zip": "^2.0.0",
```
Comment (malept) [inline package.json]: This should be alphabetized 🙈

File: script/get-cache.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  console.log('Extracting cache')
+  await extractZip(downloadFile, { dir: path.join(__dirname, '../dist') })
+
+  console.log('Cache extracted, thanks for using Electron Airlines')
+}
+
+main().catch((err) => {
+  console.log(err)
+  process.exit(1)
+})
```
Comment (malept) [inline script/get-cache.ts]: 🤔 

In theory, TypeScript 3.8 works with [top-level `await`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#top-level-await), so the `main` function shouldn't be necessary.

Comment (HashimotoYT) [reply script/get-cache.ts]: "In theory" 🙈. On practice, everything brokes after setting "module" to "esnext" or "system" what's required to enable top-level `await`. TS unable to import libraries and broken `resolveJsonModule` what's very needed for this package.

Comment (malept) [reply script/get-cache.ts]: 🤦‍♂ fair enough

**Pull request merged**